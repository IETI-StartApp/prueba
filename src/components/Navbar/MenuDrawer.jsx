import React from 'react';
import { Drawer } from '@material-ui/core/';
import { useStyles } from './styles';
import MenuItems from './MenuItems';

export default function MenuDrawer( { open, onClose} ){
    const classes = useStyles();

    return (
        <Drawer 
            className={classes.drawer}
            classes={{
                paper: classes.drawerPaper,
            }}
            anchor="left"
            variant= "temporary"
            open={open}
            onClose={onClose ? onClose : null}
        >
            <div className={classes.offset}>
                <div className={classes.menuHeader}>
                </div>
            </div>
            <MenuItems />
        </Drawer>
    );
}