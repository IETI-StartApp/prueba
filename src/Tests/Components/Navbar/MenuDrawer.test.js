import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import MenuDrawer from '../../../components/Navbar/MenuDrawer';
import { Drawer } from '@material-ui/core/';

describe('Testing menu drawer', () => {

    const handleDrawerToggle = jest.fn();

    const wrapper = shallow(<MenuDrawer open={true} onClose={handleDrawerToggle}/>)

    test('should load menu drawer', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should call handleDrawerToggle function', () => {
        const drawer = wrapper.find(Drawer).at(0);
        drawer.simulate("Close");
        expect( handleDrawerToggle ).toHaveBeenCalled();
    })
    
})
