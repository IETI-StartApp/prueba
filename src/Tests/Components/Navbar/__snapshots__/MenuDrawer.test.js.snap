// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing menu drawer should load menu drawer 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="makeStyles-drawer-6"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-7",
    }
  }
  onClose={[MockFunction]}
  open={true}
  variant="temporary"
>
  <div
    className="makeStyles-offset-10"
  >
    <div
      className="makeStyles-menuHeader-9"
    />
  </div>
  <MenuItems />
</WithStyles(ForwardRef(Drawer))>
`;
